tests/test_batch_emitter.rs

```diff
@@ -0,0 +1,76 @@
+use testcontainers::clients::Cli;
+use uuid::Uuid;
+
+use snowplow_tracker::{BatchEmitter, InMemoryEventStore, ScreenViewEvent, Tracker};
+
+mod common;
+use common::{micro_endpoint, setup, wait_for_events};
+
+#[tokio::test]
+async fn send_batches() {
+    let docker = Cli::default();
+    let (_container, micro_url) = setup(&docker);
+
+    let event_store = InMemoryEventStore::new(800, 50);
+
+    let emitter = BatchEmitter::builder()
+        .collector_url(&micro_url)
+        .event_store(event_store)
+        .build()
+        .unwrap();
+
+    let mut tracker = Tracker::new("ns", "app_id", emitter, None);
+
+    for _ in 0..800 {
+        let screenview_event = ScreenViewEvent::builder()
+            .id(Uuid::new_v4())
+            .name("a screen view")
+            .previous_name("previous screen")
+            .build()
+            .unwrap();
+
+        tracker.track(screenview_event, None).unwrap();
+    }
+
+    wait_for_events(&micro_url, "good", 800).await;
+    tracker.emitter_mut().close().unwrap();
```

Comment:
Nit: just a thought while reading this – could we add a `tracker.close()` and `tracker.flush()` convenience functions? Seems a bit cumbersome to have to go through the emitter.

Reply:
Yeah that seems sensible. I've called the closing method `tracker.close_emitter` for clarity.